File: py/2022.convert-1-d-array-into-2-d-array.py
#
# @lc app=leetcode id=2022 lang=python3
#
# [2022] Convert 1D Array Into 2D Array
#

# @lc code=start
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        if m*n != len(original):
            return []
        res = [None]*m
        c = 0
        r = 0
        elem = []
        for num in original:
            elem.append(num)
            c += 1
            if c == n:
                res[r] = elem
                r += 1
                elem = []
                c = 0
        return res

# @lc code=end

File: cpp/2022.convert-1-d-array-into-2-d-array.cpp
/*
 * @lc app=leetcode id=2022 lang=cpp
 *
 * [2022] Convert 1D Array Into 2D Array
 */
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    vector<vector<int>> construct2DArray(vector<int> &original, int m, int n)
    {
        vector<vector<int>> res;
        if (m * n != original.size())
        {
            return res;
        }
        int c = 0;
        vector<int> elem;
        for (auto num : original)
        {
            elem.push_back(num);
            ++c;
            if (c == n)
            {
                res.push_back(elem);
                elem.clear();
                c = 0;
            }
        }
        return res;
    }
};
// @lc code=end
